examples/dht_server.rs

```diff
@@ -223,36 +226,32 @@ fn main() {
             err
         });
 
-    let server: IoFuture<()> = Box::new(network);
-    let server = add_lan_sender(server, &server_obj, local_addr);
-    let server = add_server_main_loop(server, &server_obj);
-    let server = add_onion_key_refresher(server, &server_obj);
-    let server = server.join(dht_pk_handler).map(|_| ());
-    let server = server.join(lossless_handler).map(|_| ());
-    let server = server.join(lossy_handler).map(|_| ());
+    let server: IoFuture<()> = Box::new(network); // TODO: remove these boxes on rustc 1.26
+    let server: IoFuture<()> = Box::new(server.join(run_server(&server_obj)).map(|_| ()));
```

Comment:
use select instead of join

Reply:
Could you explain why you think that `select` is should be used here? With `select` you must assume that all these futures are infinite whereas `join` explicitly shows that application will be terminated when all these futures are completed.

Reply:
Because if any of the futures resolves/fails, we should return immediately. In your case we will wait until all futures are terminated.

Reply:
But in case if one future resolves with error `join` won't wait other futures.
And another argument for `join` - `select` modifies error type to [either of errors](https://docs.rs/futures/0.1.21/futures/future/struct.Select.html), so I can't just change `join` to `select`, I'll have to add `map_err` for every line here.

Reply:
A simple rule:

- use `select` if you don't have to wait for other futures.
- use `join` if you need to.

Do we need to wait for other futures?